clash survey 20220317

CamberLoid opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

- Update FOSS version of `clash` from 1.5.0 to ~~1.9.0~~ 1.10.0
- [Permission GRANTED! Dreamacro/clash/issues/2032]Introduce close-source `clash-premium` to the repository
<!-- Please input topic description here. -->

Package(s) Affected
-------------------

- `clash` version 1.5.0 -> ~~1.9.0~~ 1.10.0
- `clash-premium` ~~2022.03.19~~ 2022.03.21
<!-- Please list all package(s) affected by this topic here. -->

Security Update?
----------------

No 

Test Build(s) Done / FOSS Clash
------------------

**Primary Architectures**

FOSS Clash

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`


Test Build(s) Done / Premium Clash
------------------

**Primary Architectures**

FOSS Clash

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 / mips64le `loongson3`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

CamberLoid committed:
clash: update to 1.9.0

CamberLoid commented:
/build

CamberLoid committed:
clash-premium: new, 2022.03.19
- add pkg conflict for foss clash

WhiredPlanck commented on extra-network/clash-premium/autobuild/build:
```diff
@@ -0,0 +1,15 @@
+if [[ "${CROSS:-$ARCH}" = "loongson3" ]]; then
+    gzip -vd clash-linux-mips64le-"$VER".gz
+    install -Dvm755 clash-linux-mips64le-"$VER" "$PKGDIR"/usr/bin/clash
+fi
```
Add a blank line.

WhiredPlanck commented on extra-network/clash-premium/autobuild/defines:
```diff
@@ -0,0 +1,7 @@
+PKGDES="Close-source version of Clash with additional features."
+ABSPLITDBG=0
+PKGCONFL=clash
+FAIL_ARCH=!(amd64 arm64 loongson3)
```
Usually we mark `FAIL_ARCH` like:
```suggestion
FAIL_ARCH="!(amd64|arm64|loongson3)"
```

WhiredPlanck commented on extra-network/clash-premium/spec:
```diff
@@ -0,0 +1,7 @@
+    file::https://github.com/Dreamacro/clash/releases/download/premium/clash-linux-armv8-$VER.gz \
+    file::https://github.com/Dreamacro/clash/releases/download/premium/clash-linux-mips64le-$VER.gz"
+CHKUPDATE="html::url=https://github.com/Dreamacro/clash/releases/tag/premium;\
+    pattern=<title>Release Premium (\d\d\d\d\.\d\d\.\d\d) · Dreamacro/clash</title>"
```
Add a blank line here too.

WhiredPlanck commented on extra-network/clash/spec:
```diff
@@ -1,5 +1,5 @@
+CHKSUMS="sha256::a276b1e7247e847fe44fe336a67bdecbb442748a1e7fa01d9c1ce0d52f9168ef"
 CHKUPDATE="anitya::id=211719"
-REL=2
+REL=3
```
Remove (reset) `REL` every time after changing `VER`.

WhiredPlanck commented on extra-network/clash/autobuild/defines:
```diff
@@ -1,6 +1,8 @@
 PKGSEC=net
 PKGDEP="glibc"
 BUILDDEP="go"
 PKGDES="A rule-based tunnel for VMess, Shadowsocks, Trojan, Snell protocols"
```
When using alignment, the first and second to last words is usually connected with "and".

Although not yet encountered, a little tip for composing package description is that you can try your best to describe an application in an objective and neutral language as much as possible, instead of just copy description from upstream or other distributions.

CamberLoid committed:
clash/clashpremium: styling fix

CamberLoid committed:
clash-premium: fix FAIL_ARCH

CamberLoid committed:
clash-premium:
- spec: add architecture manipulations
- defines: remove architecture manipulations

CamberLoid committed:
clash-premium: add --force when un-gziping file

CamberLoid committed:
clash-premium: styling fix

CamberLoid commented:
clash and clash-premium is all-green and ready for review

TO-DO: 
> When using alignment, the first and second to last words is usually connected with "and".
> 
> Although not yet encountered, a little tip for composing package description is that you can try your best to describe an application in an objective and neutral language as much as possible, instead of just copy description from upstream or other distributions.

running test build

CamberLoid commented:
Test builds complete on amd64
![image](https://user-images.githubusercontent.com/22998116/159154970-dcb001c1-5510-4737-be42-bab11d84d92b.png)

CamberLoid commented:
Test build complete on loongson3
![image](https://user-images.githubusercontent.com/22998116/159155070-55bff30e-dee3-4499-acdf-b462bc5e92c0.png)

CamberLoid committed:
clash-premium: fix fail-to-build on arm64

CamberLoid committed:
clash-premium: styling fix

CamberLoid commented:
Test build complete on arm64
![image](https://user-images.githubusercontent.com/22998116/159156018-f1ce12aa-a969-4d2c-80c4-7f3d2ac39688.png)

Test build complete on ppc64el
![image](https://user-images.githubusercontent.com/22998116/159156069-f69b359d-f68f-44ae-bb1a-e1cca1be92eb.png)

CamberLoid commented:
Test build complete on riscv64
![image](https://user-images.githubusercontent.com/22998116/159156584-03254844-82f9-4297-b0b4-97d3c1e8f128.png)

MingcongBai commented on extra-network/clash/spec:
```diff
@@ -1,5 +1,5 @@
+CHKSUMS="sha256::a276b1e7247e847fe44fe336a67bdecbb442748a1e7fa01d9c1ce0d52f9168ef"
 CHKUPDATE="anitya::id=211719"
-REL=2
+REL=0
```
Drop `REL=` line when updating.

MingcongBai commented on extra-network/clash-premium/autobuild/defines:
```diff
@@ -0,0 +1,7 @@
+PKGDES="Close-source version of Clash with additional features."
+ABSPLITDBG=0
+PKGCONFL=clash
+FAIL_ARCH="!(amd64|arm64|loongson3)"
```
Did the upstream not end up offering POWER binaries?

CamberLoid replied:
https://github.com/Dreamacro/clash/issues/2032#issuecomment-1075973637
Awaiting for response

CamberLoid committed:
clash: styling fix

CamberLoid committed:
clash: update to 1.10.0

CamberLoid committed:
clash-premium: update to 2022.03.21

CamberLoid committed:
clash: Revert "clash: update to 1.10.0" ...
- clash 1.10.0 requires go 1.18+ to build. Revert to 1.9.0;
- 1.10.0 will be build in the next golang-survey

This reverts commit df7dd0bb1f572cb5e7edef355cd771d6d528f1c3.
